core/src/leader_confirmation_service.rs

```diff
@@ -125,14 +126,22 @@ mod tests {
         solana_logger::setup();
 
         let (genesis_block, mint_keypair) = GenesisBlock::new(1234);
-        let bank = Arc::new(Bank::new(&genesis_block));
+        let mut tick_hash = genesis_block.hash();
+
+        let mut bank = Arc::new(Bank::new(&genesis_block));
 
         // Move the bank up 10 slots
-        let mut tick_hash = genesis_block.hash();
-        while bank.slot_height() < 10 {
-            tick_hash = hash(&serialize(&tick_hash).unwrap());
-            bank.register_tick(&tick_hash);
+        for slot in 1..=10 {
```

Comment:
this is the only non-mechanical change I performed.  building a bank with more ticks than ticks_per_slot should probably fail (and maybe this would simplify PoHRecorder?)